New message types for ipython-kernel

thomasjm opened the pull request:
Background: I've been working on something that's essentially a new Jupyter frontend in Haskell, and I needed several more IPython message types to be supported in `ipython-kernel`. This is kind of an ongoing process, but I thought I'd get a PR out with the current changes.

gibiansky commented:
This looks good, but

1. Why is `ghc-parser` in `.gitignore`? It's a package we want in our repository.
2. There should not be more than one `stack.yaml`. There's already one for `ihaskell`, we don't need one for `ipython-kernel`, just use the same one.

thomasjm commented:
1) the ghc-parser was getting randomly updated by the build process (I was finding strings with my home directory in it). Removed from .gitignore.

2) I thought I needed it to use `ipython-kernel` as a local stack package. Removed now.

gibiansky commented on ihaskell-display/ihaskell-diagrams/IHaskell/Display/Diagrams.hs:
```diff
@@ -29,12 +30,17 @@ diagramData renderable format = do
+
   -- Convert to base64.
   imgData <- Char.readFile filename
+  putStrLn "Did readFile"
```
Debugging comments :)

gibiansky commented on ipython-kernel/src/IHaskell/IPython/Message/Parser.hs:
```diff
@@ -7,13 +7,16 @@
-import           Data.ByteString
-import           Data.Map (Map)
-import           Data.Text (Text)
+-- import           Control.Applicative ((<|>), (<$>), (<*>))
```
uncomment or remove

gibiansky commented on ipython-kernel/src/IHaskell/IPython/Message/UUID.hs:
```diff
@@ -4,7 +4,7 @@
 
 import           Control.Monad (mzero, replicateM)
-import           Control.Applicative ((<$>))
+-- import           Control.Applicative ((<$>))
```
uncomment or remove

gibiansky commented on ipython-kernel/src/IHaskell/IPython/Message/Writer.hs:
```diff
@@ -7,12 +7,12 @@ module IHaskell.IPython.Message.Writer (ToJSON(..)) where
-import qualified Data.ByteString.Lazy as L
-import qualified Data.ByteString as B
-import           Data.Text.Encoding
+-- import qualified Data.ByteString as B
```
uncomment or remove

gibiansky commented on ipython-kernel/src/IHaskell/IPython/Message/Writer.hs:
```diff
@@ -7,12 +7,12 @@ module IHaskell.IPython.Message.Writer (ToJSON(..)) where
 
 import           Data.Aeson
 import           Data.Map (Map)
+-- import           Data.Monoid (mempty)
```
uncomment or remove

gibiansky commented:
Looks good, just a few very minor comments remaining. Thanks for doing this, this is really helpful, I've actually needed this myself recently but didn't get around to it.

What are you using this for (what sort of front-end are you making)?

thomasjm commented:
Oh sorry, I didn't mean for you to review that commit yet. Emacs haskell-mode was being a little overzealous in commenting unused imports. I'll get a finalized commit out later today.

I'm making a frontend that lets you embed code in a Markdown file -- I'll post it publicly pretty soon

thomasjm commented:
Sorry @gibiansky , this fell off my radar for a while. I just updated it and addressed your comments, could we get this merged?
